service/internal/checksum/middleware_compute_input_checksum.go

```diff
@@ -130,29 +129,15 @@ func (m *computeInputPayloadChecksum) HandleFinalize(
 		})
 	}()
 
-	// If no algorithm was specified, and the operation requires a checksum,
-	// fallback to the legacy content MD5 checksum.
-	algorithm, ok, err = getInputAlgorithm(ctx)
-	if err != nil {
-		return out, metadata, err
-	} else if !ok {
-		if m.RequireChecksum {
-			checksum, err = setMD5Checksum(ctx, req)
-			if err != nil {
-				return out, metadata, computeInputHeaderChecksumError{
-					Msg: "failed to compute stream's MD5 checksum",
-					Err: err,
-				}
-			}
-			algorithm = Algorithm("MD5")
-		}
-		return next.HandleFinalize(ctx, in)
-	}
-
 	// If the checksum header is already set nothing to do.
-	checksumHeader := AlgorithmHTTPHeader(algorithm)
-	if checksum = req.Header.Get(checksumHeader); checksum != "" {
-		return next.HandleFinalize(ctx, in)
+	for _, supportedAlgorithm := range supportedAlgorithms {
+		header := AlgorithmHTTPHeader(supportedAlgorithm)
+		if cs := req.Header.Get(header); cs != "" {
```

Comment:
a corner case, but if multiple headers were to be set, do we have ranking on `supportedAlgorithms`?

Reply:
No, the purpose of this block is stop the calculation if any valid checksum header is already set

Reply:
TBH I also wonder why they don't stop on first header detected if `supportedAlgorithm` is in preference order, I can change the order and make it stop as we want

Reply:
latest SEP states If the user set any checksum header field, no calculation is needed, so the order here is not so important